### PR TITLE
Light refactoring of some transport code

### PR DESCRIPTION
**Pull request by hcaseyal**
Light refactoring of init_transport and a couple other functions for code health.

### CODE REVIEW

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                                                            FILE SIZE
 ++++++++++++++ GROWING                                                                              ++++++++++++++
  +0.0%    +200 [None]                                                                               +1.19Ki  +0.0%
      +0.0%    +200 [Unmapped]                                                                           +1.19Ki  +0.0%
      [NEW]    +192 read_channel_args(grpc_chttp2_transport*, grpc_channel_args const*, bool)::settings_    +192  [NEW]

 -------------- SHRINKING                                                                            --------------
  -0.1%     -32 src/core/ext/transport/chttp2/transport/chttp2_transport.cc                              -32  -0.1%
      [DEL]    -382 try_http_parsing(grpc_chttp2_transport*) [clone .isra.4]                                -382  [DEL]
      -9.9%     -65 grpc_core::Chttp2IncomingByteStream::Pull                                                -65  -9.9%
      -0.2%      -8 grpc_create_chttp2_transport                                                              -8  -0.2%
      -0.4%      -3 [Unmapped]                                                                                -3  -0.4%

  +0.0%    +168 TOTAL                                                                                +1.16Ki  +0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 1,955,605      Total (<)      1,955,633

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
10,844,715      Total (>)     10,844,443

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

File: src/core/ext/transport/chttp2/transport/chttp2_transport.cc [hunk 1]
```diff
@@ -2887,17 +2903,20 @@ bool Chttp2IncomingByteStream::Next(size_t max_size_hint,
   }
 }
 
+void Chttp2IncomingByteStream::EnsureStreamDecompressionCtxExists() {
```
[the remark this answers is not in the document]
Comment (hcaseyal) [reply src/core/ext/transport/chttp2/transport/chttp2_transport.cc]: Done. Thanks for the suggestion!

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                                                            FILE SIZE
 ++++++++++++++ GROWING                                                                              ++++++++++++++
  +0.0%    +168 [None]                                                                               +1.22Ki  +0.0%
      +0.0%    +200 [Unmapped]                                                                           +1.22Ki  +0.0%
      [NEW]    +192 read_channel_args(grpc_chttp2_transport*, grpc_channel_args const*, bool)::settings_    +192  [NEW]

 -------------- SHRINKING                                                                            --------------
  -0.1%     -32 src/core/ext/transport/chttp2/transport/chttp2_transport.cc                              -32  -0.1%
      [DEL]    -382 try_http_parsing(grpc_chttp2_transport*) [clone .isra.4]                                -382  [DEL]
      -9.9%     -65 grpc_core::Chttp2IncomingByteStream::Pull                                                -65  -9.9%
      -0.2%      -8 grpc_create_chttp2_transport                                                              -8  -0.2%
      -0.4%      -3 [Unmapped]                                                                                -3  -0.4%

  +0.0%    +136 TOTAL                                                                                +1.19Ki  +0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 1,954,974      Total (<)      1,955,002

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
10,844,026      Total (>)     10,843,749

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Pull request merged**